services/mlmodel/tflitecpu/tflite_cpu.go

```diff
@@ -142,7 +138,7 @@ func (m *Model) Infer(ctx context.Context, input map[string]interface{}) (map[st
 		// Fill in the output map with the names from metadata if u have them
 		// Otherwise, do output1, output2, etc.
 		for i := 0; i < len(m.model.Info.OutputTensorTypes); i++ {
-			if m.metadata != nil {
+			if m.metadata.Outputs[i].Name != "" {
```

Comment:
this should do a length check before doing a parallel array access

Reply:
Done.